### PR TITLE
TandemFoil: golden config + core physics + no-EMA

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

**Physics features on the golden config should compound with more epochs.** Physics features improved TandemFoil by 2.4% at 2 epochs (#2414). With 11 epochs at slices=64, physics features have much more training time to express their benefit. Your prior result (#2449) showed full physics + AdamW at 235.94 (EMA, 2 epochs). The golden config (slices=64, T_max=30) with physics + no-EMA + Lion should be substantially better.

## Current Baseline

| Dataset | Metric | Value | PR |
|---|---|---|---|
| TandemFoil | val_primary/surface_pressure_mae | **114.92** | #2435 (slices=64, T_max=30, EMA=True, no physics, 11 epochs) |

## Design

Two runs: core physics + golden config, one with Lion and one with AdamW (to retest optimizer preference with more epochs):

| Run | Optimizer | wandb_name |
|---|---|---|
| 1 | **Lion lr=3e-4** (matches golden config) | kaneda-golden-physics-lion |
| 2 | AdamW lr=3e-4 (retest with more epochs) | kaneda-golden-physics-adamw |

## Instructions to Student

Run **sequentially** (one at a time):

```bash
cd target/icml2026

# Run 1: Golden config + core physics + no-EMA + Lion
python train.py --dataset tandemfoil --optimizer lion --lr 3e-4 --cosine_t_max 30 \
  --no-use-ema --model_slices 64 \
  --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only \
  --asinh-pressure --residual-prediction --enable-pressure-prior-addition \
  --wandb_group kaneda-golden-physics --wandb_name kaneda-golden-physics-lion

# Run 2: Golden config + core physics + no-EMA + AdamW
python train.py --dataset tandemfoil --optimizer adamw --lr 3e-4 --cosine_t_max 30 \
  --no-use-ema --model_slices 64 \
  --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only \
  --asinh-pressure --residual-prediction --enable-pressure-prior-addition \
  --wandb_group kaneda-golden-physics --wandb_name kaneda-golden-physics-adamw
```

**Report** `val_primary/surface_pressure_mae`, per-split breakdown, and epochs completed. **Confirm 10+ epochs per run.**

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT kaneda:

## Results

### Run 1: Lion lr=3e-4 (golden config + core physics + no-EMA)

| Metric | Value | vs Baseline (114.92) |
|---|---|---|
| **val_primary/surface_pressure_mae** | **172.998** | +50.5% ❌ |
| test_primary/surface_pressure_mae | 167.231 | — |
| Epochs completed | 2 / 50 (30-min timeout) | — |

**Per-split val MAE:** single_in_dist=253.41, geom_camber_rc=202.59, geom_camber_cruise=250.01, re_rand=212.03 (epoch 1) → single_in_dist=199.43, geom_camber_rc=166.87, geom_camber_cruise=165.93, re_rand=159.75 (epoch 2)

**Per-split test MAE:** single_in_dist=179.43, geom_camber_rc=154.12, geom_camber_cruise=164.58, re_rand=170.79

- **W&B run:** 3fjtrbv6
- **Command:** `cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py --dataset tandemfoil --optimizer lion --lr 3e-4 --cosine-t-max 30 --no-use-ema --model-slices 64 --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --agent kaneda --wandb-group kaneda-golden-physics --wandb-name kaneda-golden-physics-lion`

---

### Run 2: AdamW lr=3e-4 (golden config + core physics + no-EMA)

| Metric | Value | vs Baseline (114.92) |
|---|---|---|
| **val_primary/surface_pressure_mae** | **153.097** | +33.2% ❌ |
| test_primary/surface_pressure_mae | 149.223 | — |
| Epochs completed | 2 / 50 (30-min timeout) | — |

**Per-split val MAE:** single_in_dist=206.40, geom_camber_rc=169.17, geom_camber_cruise=165.56, re_rand=157.53 (epoch 1) → single_in_dist=184.32, geom_camber_rc=151.19, geom_camber_cruise=138.83, re_rand=138.05 (epoch 2)

**Per-split test MAE:** single_in_dist=164.30, geom_camber_rc=144.77, geom_camber_cruise=138.79, re_rand=149.03

- **W&B run:** dc717g1b
- **Command:** `cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py --dataset tandemfoil --optimizer adamw --lr 3e-4 --cosine-t-max 30 --no-use-ema --model-slices 64 --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --agent kaneda --wandb-group kaneda-golden-physics --wandb-name kaneda-golden-physics-adamw`

---

## Analysis

**Both runs failed to beat the baseline (114.92), primarily due to severe epoch starvation.**

The core problem is that physics features (TE coord frame, Cp panel, asinh pressure, residual prediction, pressure prior) add ~7× overhead per epoch. At slices=64, each epoch takes ~15 minutes with physics vs ~2.7 minutes without. In the 30-min timeout window, we only complete 2 epochs vs the baseline's 11 epochs.

**Key observations:**
1. **AdamW significantly outperformed Lion** at 2 epochs (153.10 vs 173.00 val MAE, an 11.5% gap). This is the opposite of what we see without physics features, where Lion tends to win. Physics features may benefit from AdamW's more conservative update rule.
2. **Both runs were still rapidly improving** — Lion went 229.51→173.00, AdamW went 174.67→153.10 from epoch 1→2. With more epochs, convergence would likely continue.
3. **Infinity in test_geom_camber_cruise/mae_vol_p** — known early-training artifact from insufficient epochs.
4. **AdamW at 2 epochs with physics (153.10)** is already close to the non-physics no-EMA baseline at 2 epochs (197.87 from PR #2412), suggesting physics features do help per-epoch learning rate.

**The hypothesis — that physics features compound with more epochs — could not be properly tested under current timeout constraints.** The ~15 min/epoch with physics at slices=64 limits us to 2 epochs, while the no-physics baseline achieved 11 epochs.

## Suggested follow-ups

1. **Reduce slices to 32 or 48** to get more epochs with physics features — this trades resolution for more training time, which may net better results.
2. **Profile the physics feature overhead** to identify which features are most expensive. It may be possible to keep the beneficial ones while dropping the slowest.
3. **Test physics at slices=64 with a longer timeout** (if available) to actually test the compounding hypothesis with 10+ epochs.
4. **AdamW-only physics experiments** — Lion appears to underperform with physics features, so future physics experiments might default to AdamW.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2472 because ADVISOR: Closing — physics features add ~7x per-epoch overhead at slices=64, reducing from 11 epochs to 2. Neither run beats baseline (best: AdamW 153.10 vs 114.92).

**Key finding retained:** AdamW outperforms Lion with physics features (153.10 vs 173.00, -11.5%) — a reversal of the no-physics optimizer preference. Physics features improve per-epoch learning efficiency (2-epoch AdamW+physics 153.10 < old 2-epoch no-physics 197.87).

**Path forward:** Physics features at slices=32 should get ~6-10 epochs, enough to test whether physics + more training beats the golden config. Assigning you this follow-up..
